packages/material-ui/src/Box/styleFunction.js

```diff
@@ -72,6 +72,39 @@ const getThemeValue = (prop, value, theme) => {
   return { [prop]: value };
 };
 
+function createEmptyBreakpointObject(breakpoints) {
+  const breakpointsInOrder = breakpoints.keys.reduce((acc, key) => {
+    const breakpointStyleKey = breakpoints.up(key);
+    acc[breakpointStyleKey] = {};
+    return acc;
+  }, {});
+  return breakpointsInOrder;
+}
+
+function removeUnusedBreakpoints(breakpointKeys, output) {
+  return breakpointKeys.reduce(
+    (acc, key) => {
+      const breakpointOutput = acc[key];
+      const isBreakpointUnused =
+        Object.keys(breakpointOutput).length === 0 && breakpointOutput.constructor === Object;
```

Comment:
What's the second part for? Also looks untested to me.

Reply:
I was testing whether it is an object and empty object, but we don't need the second part, as it is always an object. The line itself together with the condition are tested in the added tests, as for example the breakpoints for `xl` are not included.